### DISCUSSION TITLE
Custom stl to ultrasound simulator

### DISCUSSION THREAD

**Discussion by catarinalopes21** [PlusToolkit/PlusLib#1213] [General]
I am trying to use a custom stl instead of the vessel from the example. Slicer manages to get me an ultrasound of the vessel but not of the custom stl. I made sure to switch the stl model from the PLUS file, to have the stl in RAS coordinates, to verify that there are no duplicates of any file and all files are correct. Here is what the Slicer, config file and folder look like:

![image](https://github.com/user-attachments/assets/0b2a8c04-fb4e-4057-b561-4e254a2a73c8)

![image](https://github.com/user-attachments/assets/5dcc4d46-dcbc-436a-87ea-2fa2dc048899)

![image](https://github.com/user-attachments/assets/74b87bac-927d-42a9-9e00-81564dc3f48b)

I have also tried changing parameters in PLUS and the image does change... it just does not detect the custom stl

**Comment by lassoan** [comment]
The transducer surface must touch or be slightly inside the gel block.

If it does not help then you can share your scene and PLUS device set configuration file (upload somewhere and post the link here) and then we'll have a look.

**Reply by catarinalopes21** [reply]
The transducer is inside the gel block and all I see is the outline between the gel block and the air

![image](https://github.com/user-attachments/assets/0b8c57c1-21c0-4a28-b307-db4174e19d46)

Here is the link for all the files: https://www.transfernow.net/dl/20241120jxhoM8fQ

**Reply by Sunderlandkyl** [reply]
The bone model is exported from Slicer in LPS, while the gel block is in RAS.

Try changing the "Vessel" ModelToObjectTransform to go between LPS and RAS:
```
          ModelToObjectTransform="
            -1 0 0 0
            0 -1 0 0
            0 0 1 0
            0 0 0 1"
```

You could also just export the bone model again, making sure to write it in RAS coordinates.

**Reply by catarinalopes21** [reply]
It worked! 

![image](https://github.com/user-attachments/assets/28d32c75-4b60-4898-883b-c32edb421462)

However, how do I make it more similar to a real bone surface ultrasound?

**Reply by catarinalopes21** [reply]
By this I mean having the fat/muscle-like image of the ultrasound in the place where the gel appears in the image. So something like this:

![image](https://github.com/user-attachments/assets/5518c1d1-6418-4083-af66-c86fdc050159)

**Reply by lassoan** [reply]
It is great that it works now. The original question is resolved.
Please start a new discussion for each topic.
